### PR TITLE
enhance(runtime): refactor runtime dependencies with user original sequence

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
- related: https://github.com/star-whale/starwhale/issues/780
- [asciicast](https://asciinema.org/a/534239)

## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/runtime/model.py [hunk 1]
```diff
@@ -177,37 +183,289 @@ def __str__(self) -> str:
...
+
+    @property
+    def kind(self) -> DependencyType:
+        return DependencyType.NATIVE_FILE
+
+    def _do_install(self, workdir: Path, mode: str) -> None:
+        # TODO: support native file pre-hook, post-hook
+        for d in self.deps:
+            if d["dest"].startswith("/") and in_container():
+                logger.warning(f"NOTICE! dest dir: {d['dest']}")
```
Comment (jialeicui) [inline client/starwhale/core/runtime/model.py:228]: Get the absolute path and then compare the commonprefix with work dir?
`dest` might be "../../../"

File: client/starwhale/core/runtime/model.py [hunk 2]
```diff
@@ -177,37 +183,289 @@ def __str__(self) -> str:
...
+                _dest = Path(d["dest"])
+            else:
+                if mode in (PythonRunEnv.CONDA, PythonRunEnv.VENV):
+                    _root = workdir / "export" / mode
+                else:
+                    _root = workdir
```
Comment (jialeicui) [inline client/starwhale/core/runtime/model.py:234]: It seems that this line is unreachable.

File: client/starwhale/core/runtime/model.py (hunk 1 above)
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py:228]: ok

File: client/starwhale/core/runtime/model.py (hunk 2 above)
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py]: fixed

File: client/starwhale/core/runtime/model.py (hunk 1 above)
Comment (tianweidut) [reply client/starwhale/core/runtime/model.py]: fixed

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**